csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -200,6 +214,7 @@ public ArrowStreamWriter(Stream baseStream, Schema schema, bool leaveOpen, IpcOp
 
             _fieldTypeBuilder = new ArrowTypeFlatbufferBuilder(Builder);
             _options = options ?? IpcOptions.Default;
+            _dictionaryMemo = new DictionaryMemo();
```

Comment:
Can this be lazy-initialized until it is needed? That way every ArrowStreamWriter doesn't need to create it, if they don't use Dictionaries.